cluster-autoscaler/FAQ.md

```diff
@@ -807,71 +807,139 @@ The following startup parameters are supported for cluster autoscaler:
+| `scale-down-non-empty-candidates-count` | Maximum number of non empty nodes considered in one iteration as candidates for scale down with drain.Lower value means better CA responsiveness but possible slower scale down latency.Higher value can affect CA performance with big clusters (hundreds of nodes).Set to non positive value to turn this heuristic off - CA will not limit the number of nodes it considers. | 30 |
+| `scale-down-simulation-timeout` | How long should we run scale down simulation. | 30s |
+| `scale-down-unneeded-time` | How long a node should be unneeded before it is eligible for scale down | 10m0s |
+| `scale-down-unready-enabled` | Should CA scale down unready nodes of the cluster | true |
+| `scale-down-unready-time` | How long an unready node should be unneeded before it is eligible for scale down | 20m0s |
+| `scale-down-utilization-threshold` | The maximum value between the sum of cpu requests and sum of memory requests of all pods running on the node divided by node's corresponding allocatable resource, below which a node can be considered for scale down | 0.5 |
+| `scale-up-from-zero` | Should CA scale up when there are 0 ready nodes. | true |
+| `scan-interval` | How often cluster is reevaluated for scale up or down | 10s |
+| `scheduler-config-file` | scheduler-config allows changing configuration of in-tree scheduler plugins acting on PreFilter and Filter extension points |  |
+| `skip-headers` | If true, avoid header prefixes in the log messages |  |
+| `skip-log-headers` | If true, avoid headers when opening log files (no effect when -logtostderr=true) |  |
+| `skip-nodes-with-custom-controller-pods` | If true cluster autoscaler will never delete nodes with pods owned by custom controllers | true |
+| `skip-nodes-with-local-storage` | If true cluster autoscaler will never delete nodes with pods with local storage, e.g. EmptyDir or HostPath | true |
+| `skip-nodes-with-system-pods` | If true cluster autoscaler will never delete nodes with pods from kube-system (except for DaemonSet or mirror pods) | true |
+| `startup-taint` | Specifies a taint to ignore in node templates when considering to scale a node group (Equivalent to ignore-taint) | [] |
+| `status-config-map-name` | Status configmap name | "cluster-autoscaler-status" |
+| `status-taint` | Specifies a taint to ignore in node templates when considering to scale a node group but nodes will not be treated as unready | [] |
+| `stderrthreshold` | logs at or above this threshold go to stderr when writing to files and stderr (no effect when -logtostderr=true or -alsologtostderr=true) | 2 |
+| `unremovable-node-recheck-timeout` | The timeout before we check again a node that couldn't be removed before | 5m0s |
+| `user-agent` | User agent used for HTTP calls. | "cluster-autoscaler" |
+| `v` | number for the log level verbosity |  |
```

Comment:
@rainfd, curious to know about `v`,  there is no such flag for CA under [main.go](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/main.go)

Reply:
`v` and `vmodule` come from klog, Is it necessary to remove it?

Reply:
@rainfd, This table contains parameters(flags) that are supported for CA, so mentioning the Klog flag will create confusion for the users.

Reply:
>  Is it necessary to remove it?

So it will be good if the Klog flags can be removed from this table.

Reply:
> > Is it necessary to remove it?
> 
> So it will be good if the Klog flags can be removed from this table.

@Shubham82  cluster-autoscaler also use `klog.V(x).Info()` to mark log of different levels, this usage is common in the k8s community, so I don't think it really matters

Reply:
> @Shubham82 cluster-autoscaler also use klog.V(x).Info() to mark log of different levels, this usage is common in the k8s community, so I don't think it really matters

@rainfd, I know its usage, but I am concerned with this table, as it is dedicated to only the CA flags.  so I think only CA dedicated flags can be mentioned here, other than that changes look good to me. IMO If somehow we can remove these flags, that will be good.

Let's wait for other member's opinions.

Reply:
cc @gjtempleton.

Reply:
Adding `--v` flag to cluster-autoscaler binary increases log verbosity and helps in debugging. I don't think the end user cares where it comes from.

Reply:
We should write docs with both technical and non-technical users in mind.

Reply:
Hi @vadasambar, My only concern here is that the parameters mentioned here are for CA (comes with CA) and `--v` is for log verbosity, not a specific parameter to CA.

The following also mentioned in [FAQ](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-are-the-parameters-to-ca) before the parameter table:

> The following startup parameters are supported for cluster autoscaler

Reply:
If all members are ok with it (mention --v in the CA parameter table) if it somehow helps end users/users then I'm also ok with it.

Reply:
cc @gjtempleton, WDYT?

Reply:
Users can set it on klog and it affects the logging behaviour of the CA, I'd list it.